index.js

```diff
@@ -0,0 +1,142 @@
+axios.defaults.baseURL = 'https://trektravel.herokuapp.com/trips';
+//
+// Status Management
+//
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+  $('#status-message').removeClass('alert alert-danger')
+  $('#status-message').addClass('alert alert-success')
+};
+const errorStyling = () => {
+  $('#status-message').removeClass('alert alert-success')
+  $('#status-message').addClass('alert alert-danger')
+}
+
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+
+//
+// Loading Trips
+//
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+  $('h2').removeClass('hidden')
+
+  axios.get()
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      response.data.forEach((trip) => {
+        const listItem = $(`<li>${trip.name}</li>`);
+        tripList.append(listItem);
+        listItem.click(showTripDetails(trip.id));
+      });
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      errorStyling();
+      console.log(error);
+    });
+};
+
+//Load Trip Details
+const showTripDetails = (tripId) => {
+  const loadTripDetails = () => {
+    reportStatus('Loading trip details...');
+    $('.details').removeClass('hidden')
+    
+    const tripDetail = $('#trip-detail');
+    tripDetail.empty();
+
+    axios.get(`${tripId}`)
+    
+    .then((response) => {
+      console.log(response);
+      let trip = response.data
+      reportStatus(`Successfully loaded ${trip.name} details`);   
+        tripDetail.append(`<li><span> Name:</span> ${trip.name}</li>`);
+        tripDetail.append(`<li><span> Continet:</span> ${trip.continent}</li>`);
+        tripDetail.append(`<li><span> Category:</span> ${trip.category}</li>`);
+        tripDetail.append(`<li><span> Weeks:</span> ${trip.weeks}</li>`);
+        tripDetail.append(`<li><span> Cost:</span> $${trip.cost}</li>`);
+        tripDetail.append(`<li><span> About:</span> ${trip.about}</li>`);
+
+      const reserve = reservationHandler(tripId);
+      $('#reservation-form').submit(reserve);
```

Comment:
When reserving a second trip, your site makes two POST requests. To reproduce:
1. Open page
1. Open chrome dev tools to the "Network" tab
1. Click See All Trips
1. Click on the first trip (Cairo to Zanzibar)
1. Make a reservation and submit
1. Click on the second trip (Everest Base Camp Trek)
1. Make a reservation and submit
1. Observe that on submit, 2 POST requests were made to trek API's /reservations endpoint

The reason for this is that you reuse the same reservation form for each trip, but re-attach the `submit` handler every time you show details for a trip.